src/main/java/bio/terra/janitor/db/TrackedResource.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.janitor.db;
+
+import bio.terra.generated.model.CloudResourceUid;
+import com.google.auto.value.AutoValue;
+import java.time.Instant;
+
+/**
+ * A single tracked resource for persistence.
+ *
+ * <p>The Janitor service attempts to cleanup resources that it is tracking once they expire. This
+ * class describes a single resource being tracked.
+ */
+@AutoValue
+public abstract class TrackedResource {
```

Comment:
I would assume once we have the get endpoint, we would have a TrackedResource in openAPI. Seems WM and some DataRepo use that directly in DB. Then do we still need this? Or we just define it in openApi directly?

Reply:
I'd rather not mix layers of serialization. There may be things we want to add to the request API that we don't want in the database and vice versa. Locking different serialization layers has caused me pain in the past. It's easier to write thin conversion code to take `input request format` -> `internal representation` -> `database representation`.
The exception we have to that right now is CloudResourceUid, which I honestly don't feel great about, but it's supposed to be an external "standard" for us, so it's perhaps ok.